### PR TITLE
[Feature] Add support for view controllers

### PR DESCRIPTION
**Pull request by willer88**
* **Commits:**

* Add StepsContainerViewController class

### CODE REVIEW

File: step-control-ios/StepsContainerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  func lastStepController() {
+    guard let container = parent as? StepsContainerViewController else { return }
+    container.lastPage()
+  }
+}
+
+public class StepsContainerViewController: UIViewController {
+
+  // MARK: Properties
+  public var stepControl: StepControl = StepControl()
```
Comment (fernandodelafuente) [inline step-control-ios/StepsContainerViewController.swift:26]: `public var stepControl = StepControl()` type can be infered

File: step-control-ios/StepsContainerViewController.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+// MARK: StepControlDataSource
+extension StepsContainerViewController: StepControlDataSource {
+  
+    public func numberOfItems(viewPager: StepControl) -> Int {
+    return viewSteps.count
+  }
+  
+    public func viewAtIndex(viewPager: StepControl, index: Int, view: UIView?) -> UIView {
+    return viewSteps[index]
+  }
```
Comment (maite-manana) [inline step-control-ios/StepsContainerViewController.swift:92]: fix indentation

File: step-control-ios/StepsContainerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      stepIndex += 1
+      stepControl.scrollToPage(index: stepIndex)
+    }
+  }
+  
+  public func lastPage() {
+    if stepIndex > 1 {
+      stepIndex -= 1
+      stepControl.scrollToPage(index: stepIndex)
+    }
```
Comment (maite-manana) [inline step-control-ios/StepsContainerViewController.swift:49]: change to a guard?

File: step-control-ios/StepsContainerViewController.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    container.lastPage()
+  }
+}
+
+public class StepsContainerViewController: UIViewController {
+
+  // MARK: Properties
+  public var stepControl: StepControl = StepControl()
+  private var viewSteps: [UIView] = []
+  private var stepIndex: Int = 1
```
Comment (fernandodelafuente) [inline step-control-ios/StepsContainerViewController.swift:28]: `private var stepIndex = 1` idem

File: step-control-ios/StepsContainerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    steps.forEach { configureStep(controller: $0) }
+    
+    configureStepControl()
+  }
+  
+  public func nextPage() {
+    if stepIndex < viewSteps.count {
+      stepIndex += 1
+      stepControl.scrollToPage(index: stepIndex)
+    }
```
Comment (fernandodelafuente) [inline step-control-ios/StepsContainerViewController.swift:42]: prefer guard when possible

File: step-control-ios/StepsContainerViewController.swift (hunk 3 above)
Comment (fernandodelafuente) [inline step-control-ios/StepsContainerViewController.swift:49]: idem guard

File: step-control-ios/StepsContainerViewController.swift [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  private func configureStepControlLayout() {
+    stepControl.translatesAutoresizingMaskIntoConstraints = false
+    view.addSubview(stepControl)
+    
+    let constraints = [stepControl.topAnchor.constraint(equalTo: view.topAnchor),
+                       stepControl.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+                       stepControl.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                       stepControl.trailingAnchor.constraint(equalTo: view.trailingAnchor)]
+    
+    NSLayoutConstraint.activate(constraints)
```
Comment (fernandodelafuente) [inline step-control-ios/StepsContainerViewController.swift:79]: if you are not going to use `constraints` let prefer this:

```
NSLayoutConstraint.activate([
  stepControl.topAnchor.constraint(equalTo: view.topAnchor),
  stepControl.bottomAnchor.constraint(equalTo: view.bottomAnchor),
  stepControl.leadingAnchor.constraint(equalTo: view.leadingAnchor),
  stepControl.trailingAnchor.constraint(equalTo: view.trailingAnchor)
])
```

**Pull request closed without merging**